raft.c

```diff
@@ -231,6 +231,40 @@ static void executeLogEntry(RedisRaftCtx *rr, raft_entry_t *entry, raft_index_t
     }
 }
 
+static void raftSendNodeShutdown(raft_node_t *raft_node)
+{
+    if (!raft_node) {
+        return;
+    }
+
+    Node *node = raft_node_get_udata(raft_node);
+    if (!node) {
+        return;
+    }
+
+    char tmp[32];
+    snprintf(tmp, sizeof(tmp), "%d", raft_node_get_id(raft_node));
+
+    char* argv[2] = {
+        "RAFT.NODESHUTDOWN",
+        tmp
+    };
+
+    size_t argvlen[2] = {
+        strlen("RAFT.NODESHUTDOWN"),
+        strlen(tmp)
+    };
+
+    if (!ConnIsConnected(node->conn)) {
+        NODE_TRACE(node, "not connected, state=%s", ConnGetStateStr(node->conn));
+        return;
+    }
+
+    if (redisAsyncCommandArgv(ConnGetRedisCtx(node->conn), NULL,
```

Comment:
Any reason to use `redisAsyncCommandArgv` and not just `redisAsyncCommand` here? Might be shorter and simpler.

Reply:
I don't remember, probably unfamiliarity with the hiredis API caused that. Fixed.